test/e2e/common.go

```diff
@@ -0,0 +1,370 @@
+package e2e
+
+import (
+	"context"
+	"fmt"
+	. "github.com/onsi/gomega"
+	clusterclient "github.com/open-cluster-management/api/client/cluster/clientset/versioned"
+	operatorclient "github.com/open-cluster-management/api/client/operator/clientset/versioned"
+	workv1client "github.com/open-cluster-management/api/client/work/clientset/versioned"
+	clusterv1 "github.com/open-cluster-management/api/cluster/v1"
+	operatorapiv1 "github.com/open-cluster-management/api/operator/v1"
+	workapiv1 "github.com/open-cluster-management/api/work/v1"
+	"github.com/open-cluster-management/registration-operator/pkg/helpers"
+	certificatesv1beta1 "k8s.io/api/certificates/v1beta1"
+	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1beta1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/util/retry"
+)
+
+const (
+	eventuallyTimeout            = 60 // seconds
+	eventuallyInterval           = 1  // seconds
+	clusterManagerNamespace      = helpers.ClusterManagerNamespace
+	klusterletDefaultNamespace   = helpers.KlusterletDefaultNamespace
+	bootstrapHubKubeConfigSecret = helpers.BootstrapHubKubeConfigSecret
+	hubRegistrationDeployment    = "cluster-manager-registration-controller"
+	hubWebhookDeployment         = "cluster-manager-registration-webhook"
+	operatorNamespace            = "open-cluster-management"
+	klusterletOperator           = "klusterlet"
+)
+
+var (
+	kubeClient         kubernetes.Interface
+	clusterCfg         *rest.Config
+	operatorClient     operatorclient.Interface
+	clusterClient      clusterclient.Interface
+	workClient         workv1client.Interface
+	bootstrapHubSecret *corev1.Secret
+)
+
+func createKlusterlet(name, clusterName, agentNamespace string) (realClusterName string) {
+	var klusterlet = &operatorapiv1.Klusterlet{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
+		},
+		Spec: operatorapiv1.KlusterletSpec{
+			RegistrationImagePullSpec: "quay.io/open-cluster-management/registration",
+			WorkImagePullSpec:         "quay.io/open-cluster-management/work",
+			ExternalServerURLs: []operatorapiv1.ServerURL{
+				{
+					URL: "https://localhost",
+				},
+			},
+			ClusterName: clusterName,
+			Namespace:   agentNamespace,
+		},
+	}
+
+	Expect(name).NotTo(BeEmpty())
```

Comment:
it's better to check input values before doing anything with them; this should be at the top of this method

Reply:
done